### PR TITLE
fix: update dependencies, userinfo endpoint api, show roles

### PR DESCRIPTION
**Pull request by peintnermax**
Closes https://github.com/zitadel/zitadel/issues/5188

- [x] calls userinfo endpoint and shows user data
- [x] show roles from userinfo
- [x] update dependencies

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,13 +71,15 @@ ZitadelProvider({
 ```
 
 We recommend using the Authentication Code flow secured by PKCE for the Authentication flow.
-To be able to connect to ZITADEL, navigate to your Console Projects, create or select an existing project and add your app selecting WEB, then PKCE, and then add `http://localhost:3000/api/auth/callback/zitadel` as redirect url to your app.
+To be able to connect to ZITADEL, navigate to your Console Projects, create or select an existing project and add your app selecting WEB, then CODE, and then add `http://localhost:3000/api/auth/callback/zitadel` as redirect url to your app.
```
Comment (eliobischof) [inline README.md:74]: Why no PKCE anymore?

File: pages/api/auth/[...nextauth].tsx [hunk 2]
```diff
@@ -1,12 +1,84 @@
...
+      session.user = {
+        id: user?.id,
+        email: user?.email,
+        image: user?.image,
+        name: user?.name,
+        loginName: user?.loginName,
+      };
+      session.clientId = process.env.ZITADEL_CLIENT_ID;
+      session.error = tokenError;
+      return session;
```
Comment (eliobischof) [inline pages/api/auth/[...nextauth].tsx:81]: Also here, does it make sense to split different configs up into separate files?

File: README.md (hunk 1 above)
Comment (peintnermax) [reply README.md:74]: cause this is needed if you need a secret from zitadel

File: pages/api/auth/[...nextauth].tsx [hunk 3]
```diff
@@ -1,12 +1,84 @@
...
+    };
+  } catch (error) {
+    console.error('Error during refreshAccessToken', error);
+
+    return {
+      ...token,
+      error: 'RefreshAccessTokenError',
+    };
+  }
+}
```
[the remark this answers is not in the document]
Comment (peintnermax) [reply pages/api/auth/[...nextauth].tsx:30]: I would onestly keep all this auth logic in this file

File: pages/api/auth/[...nextauth].tsx (hunk 2 above)
Comment (peintnermax) [reply pages/api/auth/[...nextauth].tsx:81]: would not do

**Review by eliobischof — APPROVED** [review]

**Pull request merged**